services/QuillLMS/app/helpers/content_hubs_helper.rb

```diff
@@ -0,0 +1,295 @@
+# frozen_string_literal: true
+
+module ContentHubsHelper
+
+  def unit_activities_include_content_activities?(unit_activities, content_activity_ids)
+    unit_activities.where(activity_id: content_activity_ids).any?
+  end
+
+  def unit_activities_include_social_studies_activities?(unit_activities)
+    activity_ids = world_history_1200_to_present.map { |unit_template| unit_template[:activities].map { |act| act[:activity_id] } }.flatten
```

Comment:
`world_history_1200_to_present` is not nested, so I don't think that `flatten` does anything here.  Though it doesn't hurt anything, so leaving it might be fine?  I half-suspect that you might have meant to call `compact` instead, though, since that's the one that eliminates `nil` values from the array, and we know there will be some.  Though, again, I don't think that having `nil` values in this array hurts anything.  So maybe nothing needs to be changed here.

Reply:
There's a double map, through the unit templates and their nested activities, so we have to call `flatten` to avoid getting an array of arrays of activity ids back here. You're right that it does contain some `nil` values, but you're also correct that it doesn't make a difference for what we're doing here.